R/truncation.R

```diff
@@ -0,0 +1,130 @@
+#' Adjust or subset a line list to account for right-truncation
+#'
+#' @description
+#' Adjust or subset the line list `<data.frame>` by either changing dates that
+#' are within the right-truncation window to `NA` or removing cases that have
+#' occurred within a delay period. This occurs to recent cases or outcomes
+#' that are not yet recorded in the line list and will be revised upwards. See
+#' `truncation_event` argument documentation for different types of truncation.
+#'
+#' @details
+#' The truncation window is sampled for each individual and is specified by a
+#' (random) number generating function supplied to the `delay`
+#' argument. Therefore, if this function has variability
+#' (e.g. a probability distribution) it will not be a fixed time window for
+#' all individuals. It is possible that someone with a more recent date of
+#' onset is kept but someone with a less recent date of onset is removed due
+#' to the variability in the reporting delays. See examples for variable or
+#' fixed `delay` functions.
+#'
+#' @param linelist A `<data.frame>` output from [sim_linelist()].
+#' @param delay A `function` (either anonymous or predefined) that
+#' has a single argument and generates random numbers given a probability
+#' distribution. The function must return a vector of real numbers for
+#' representing the reporting delay between the right-truncation point and the
+#' cut off date (`max_date`).
+#'
+#' Default is a random number generator that is lognormally
+#' distributed ([rlnorm()]) delay with parameters `meanlog = 0.58` and
+#' `sdlog = 0.47`, which corresponds to a mean of 2 and a standard deviation
+#' of 1.
+#' @param max_date A maximum date to cut off the outbreak and apply reporting
+#' delays. Default is `NULL`, so the maximum date will be automatically
+#' calculated as the date at the end of the outbreak (i.e. date of the last
+#' outcome).
+#' @param truncation_event A `character` string with which event in the line
+#' list the right-truncation should apply to. The default is `"reporting"` for
+#' the reporting delay, which is likely the most common form of right-truncation
+#' in real-time outbreak data. When `truncation_event = "reporting"` if a date
+#' of reporting (`$date_reporting`) is more recent than the sampled truncation
+#' time then the individual (row) is removed from the line list. If the date of
+#' reporting is less recent than the sampled truncation time but either the
+#' date of hospitalisation (`$date_admission`) or date of outcome
+#' (`$date_outcome`) is more recent than the sampled truncation time then these
+#' dates are converted from `Date`s to `NA`s.
+#'
+#' The other options for `truncation_event` are `"onset"`, `"admission"`
+#' or `"outcome"`.  If these are chosen then if the truncation point is more
```

Comment:
sorry again but at this point I'm lost. There is quite a bit of terminology with truncation points, truncation events, truncation times, right-truncation windows, delay periods. Perhaps define these all in the function definition and then refer to them in the argument description?